### ISSUE TITLE
Add support for the Trigger API on custom page fields {variable}

### ISSUE THREAD

**Issue by jeremymouton** [rainlab/pages-plugin#236]
In YML and PHP, we can define a trigger by doing this:

```yml
my_field:
  label: Link
  span: right
  trigger:
     action: show
     field: my_other_field
     condition: checked
```

Is there a way to pass a trigger to a custom form element?

Such as 

```
{variable
  name="my_field"
  label="Link"
  trigger-action="show"
  trigger-field="my_other_field"
  trigger-condition="checked"
  span="right"}
{/variable}
```

**Comment by daftspunk** [comment]
Not sure the syntax fields supports this at the moment, we would need to introduce it.

**Comment by LukeTowers** [comment]
Closing as it has been over a month since any activity on this occurred.

**Comment by rangrage** [comment]
@daftspunk I would like to request this functionality as well please.

**Comment by LukeTowers** [comment]
@rangrage feel free to make a PR to add it. See https://github.com/octobercms/library/blob/master/src/Parse/Syntax/FieldParser.php for where it happens in the codebase to get an idea of where to start.

**Comment by developerabdd** [comment]
@LukeTowers @daftspunk I, too, would very much like to see this functionality (or something like it) implemented.! There have been way too many situations where this would have saved us a few headaches!

**Comment by LukeTowers** [comment]
@developerabdd @rangrage if either of you are interested in sponsoring me to add this support then shoot me an email at octobercms@luketowers.ca or send me a PM on Slack

**Comment by jeremymouton** [comment]
Using "Grouped repeaters" works as a workaround for now: https://octobercms.inetis.ch/advanced-repeaters-static-pages-plugin

**Comment by rangrage** [comment]
@jeremymouton thank you for this. It looks like a good work around for now. :)

**Comment by daftspunk** [comment]
Hey @jeremymouton 

Custom fields are handled mainly by Tailor now. The "demo" theme shipped in v3.4 has a "Wiki" example that effectively does the same thing.
